zkevm-circuits/src/evm_circuit/execution/callop.rs

```diff
@@ -102,12 +101,31 @@ impl<F: Field> ExecutionGadget<F> for CallGadget<F> {
         // Lookup values from stack
         cb.stack_pop(gas_word.expr());
         cb.stack_pop(callee_address_word.expr());
-        cb.stack_pop(value.expr());
-        cb.stack_pop(cd_offset.expr());
-        cb.stack_pop(cd_length.expr());
-        cb.stack_pop(rd_offset.expr());
-        cb.stack_pop(rd_length.expr());
-        cb.stack_push(is_success.expr());
+        // Save the current stack pointer offset to increase outside of
+        // ConstraintBuilder. Since opcode `CALL` has an additional stack pop
+        // `value` excluded by opcode `STATICCALL`.
+        let stack_offset = cb.stack_pointer_offset();
+        cb.condition(is_call.expr(), |cb| {
+            cb.stack_lookup(false.expr(), stack_offset.expr(), value.expr())
+        });
+        let stack_offset = stack_offset.expr() + is_call.expr();
+        cb.stack_lookup(false.expr(), stack_offset.expr(), cd_offset.expr());
+        cb.stack_lookup(
+            false.expr(),
+            stack_offset.expr() + 1.expr(),
+            cd_length.expr(),
+        );
+        cb.stack_lookup(
+            false.expr(),
+            stack_offset.expr() + 2.expr(),
+            rd_offset.expr(),
+        );
+        cb.stack_lookup(
+            false.expr(),
+            stack_offset.expr() + 3.expr(),
+            rd_length.expr(),
+        );
+        cb.stack_lookup(true.expr(), stack_offset + 3.expr(), is_success.expr());
```

Comment:
These can be replaced with `stack_pop` and `stack_push` right?

Reply:
No, so the discussion here: https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/801#discussion_r997670440

Reply:
Okay! Let's wait for #837 to be merged, and then re-run the build for this PR?